velox/functions/sparksql/aggregates/tests/SumAggregationTest.cpp

```diff
@@ -26,7 +30,110 @@ class SumAggregationTest : public SumTestBase {
  protected:
   void SetUp() override {
     SumTestBase::SetUp();
-    registerAggregateFunctions("spark_");
+    registerAggregateFunctions("spark_", true);
+  }
+
+ protected:
+  // Check global partial agg overflow, and final agg output null.
+  void decimalGlobalSumOverflow(
+      const std::vector<std::optional<int128_t>>& input,
+      const std::vector<std::optional<int128_t>>& output) {
+    const TypePtr type = DECIMAL(38, 0);
+    auto in = makeRowVector({makeNullableFlatVector<int128_t>({input}, type)});
+    auto expected =
+        makeRowVector({makeNullableFlatVector<int128_t>({output}, type)});
+    testAggregations(
+        {in},
+        {},
+        {"spark_sum(c0)"},
+        {expected},
+        /*config*/ {},
+        /*testWithTableScan*/ false);
+    testAggregationsWithCompanion(
+        {in},
+        [](auto& /*builder*/) {},
+        {},
+        {"spark_sum(c0)"},
+        {{type}},
+        {},
+        {expected},
+        {});
+  }
+
+  // Check group by partial agg overflow, and final agg output null.
+  void decimalGroupBySumOverflow(
+      const std::vector<std::optional<int128_t>>& input) {
+    const TypePtr type = DECIMAL(38, 0);
+    auto in = makeRowVector(
+        {makeFlatVector<int32_t>(20, [](auto row) { return row % 10; }),
+         makeNullableFlatVector<int128_t>(input, type)});
+    auto expected = makeRowVector(
+        {makeFlatVector<int32_t>(10, [](auto row) { return row; }),
+         makeNullableFlatVector<int128_t>(
+             std::vector<std::optional<int128_t>>(10, std::nullopt), type)});
+    testAggregations(
+        {in},
+        {"c0"},
+        {"spark_sum(c1)"},
+        {expected},
+        /*config*/ {},
+        /*testWithTableScan*/ false);
+    testAggregationsWithCompanion(
+        {in},
+        [](auto& /*builder*/) {},
+        {"c0"},
+        {"spark_sum(c1)"},
+        {{type}},
+        {"c0", "a0"},
+        {expected},
+        {});
+  }
+
+  template <typename TIn, typename TOut>
+  void decimalSumAllNulls(
+      const std::vector<std::optional<TIn>>& input,
+      const TypePtr& inputType,
+      const std::vector<std::optional<TOut>>& output,
+      const TypePtr& outputType) {
+    std::vector<RowVectorPtr> vectors;
+    VectorPtr inputDecimalVector;
+    if constexpr (std::is_same_v<int64_t, TIn>) {
+      inputDecimalVector = makeNullableFlatVector<int64_t>(input, inputType);
+    } else {
+      inputDecimalVector = makeNullableFlatVector<int128_t>(input, inputType);
+    }
```

Comment:
nit: No need for this check. Just `inputDecimalVector = makeNullableFlatVector<TIn>(input, inputType);` should be equivalent.